### DISCUSSION TITLE
Renovate updated VSCode compatibility engine from semantic to calendar versioning

### DISCUSSION THREAD

**Discussion by kieran-ryan** [renovatebot/renovate#26356] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

With automerge enabled, Renovate updated the [compatibility engine version of VSCode](https://code.visualstudio.com/api/working-with-extensions/publishing-extension#visual-studio-code-compatibility) for an extension from semantic to calendar versioning: `^1.82.0` to `^2023.0.0`. As VSCode uses semantic versioning, the calendar version will not be recognized and will also always exceed the expected semantic version - so the extension will never run.

```console
2023-12-17 20:39:40.976 [info] Started local extension host with pid 99070.
2023-12-17 20:39:41.903 [error] [/Users/<username>/cucumber/vscode]: Extension is not compatible with Code 1.85.0. Extension requires: ^2023.0.0.
```

Is it possible that there was a yanked tag that was created on microsoft/vscode - as no tag currently exists with that version? I have been unable to identify where the calendar version came from; does Renovate keep records of this detail, or is there a way of verifying the origin? Would Renovate still apply this change if I revert?

See cucumber/vscode#190 for more detail and the commit introduced by Renovate: [99e5f3b](https://github.com/cucumber/vscode/commit/99e5f3b05d1f5bc89f8bd710a41837e5f5f2b5c8).

### Relevant debug logs

The values pulled from the latest manifest appear as the below.

```console
        [
          ...
          {
            "currentValue": "^2023.0.0",
            "currentVersion": null,
            "datasource": "github-tags",
            "depName": "vscode",
            "depType": "engines",
            "lockedVersion": null,
            "packageName": "microsoft/vscode",
            "prettyDepType": "engine",
            "registryUrl": "https://github.com",
            "skipReason": "invalid-version",
            "sourceUrl": "https://github.com/microsoft/vscode",
            "versioning": "npm",
            "warnings": [],
            "updates": []
          }
        ],
        "extractedConstraints": {
          "npm": "<9",
          "vscode": "^2023.0.0"
        },
```

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment, marked as answer]
Technical quibble: 

The tags are here: https://github.com/microsoft/vscode/tags

I don't see any 2023 one, however Renovate is not an LLM and does not _hallucinate_ strings. I wonder if a mistaken tag was released and removed.

BTW I recommend disabling automerge by default for major releases, which would have prevented this

**Reply by kieran-ryan** [reply]
Cheers for reply @rarkins.

Aligns with assumption mentioned above that VSCode yanked a tag - though was yet unable to locate evidence in tag-based workflows or git history. However, despite unchanged configuration, this is the first time Renovate has updated the VSCode constraint - granted this would be the first release classified "major". In a previous discussion, a user outlined that [Renovate does not update VSCode engines](https://github.com/renovatebot/renovate/discussions/24365).

Is it the case that Renovate updates the VSCode constraint for major releases only? I have been unable to locate the relevant code. In any case, I intend to disable all automerges of this constraint within the affected repository.

[Cucumber organisation default renovate configuration](https://github.com/cucumber/renovate-config/blob/0dd3937e2d04afc06d4fd9ad9802fde1276a3e8b/default.json):

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base",
    ":preserveSemverRanges",
    ":rebaseStalePrs",
    ":disableDependencyDashboard",
    ":automergeBranch",
    ":automergePatch",
    ":automergeMinor",
    "github>cucumber/renovate-config:gemspec",
    "github>cucumber/renovate-config:disable-perl"
  ],
  "prHourlyLimit": 0
}
```

[Cucumber VSCode extension renovate configuration](https://github.com/cucumber/vscode/blob/7fcb66a005aca3e7324f2a59fae62b9e6faff97c/.github/renovate.json):

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "github>cucumber/renovate-config",
    ":automergeMajor"
  ]
}
```

**Reply by rarkins** [reply]
The other user was mistaken. Renovate simply doesn't change ranges unless it needs to, which is a sensible default
